clients/src/main/java/org/apache/kafka/common/config/ConfigDef.java

```diff
@@ -99,10 +103,16 @@ public ConfigDef define(ConfigKey key) {
         if (configKeys.containsKey(key.name)) {
             throw new ConfigException("Configuration " + key.name + " is defined twice.");
         }
+        configKeys.put(key.name, key);
+        for (String name: key.deprecatedNames) {
+            if (configKeys.containsKey(name)) {
+                throw new ConfigException("Configuration " + key.name + " is defined twice.");
```

Comment:
Since it's tied to the deprecated name, the error message should probably include both the deprecated and new name. Otherwise it'll be more difficult to track down the source of the conflict.